### PR TITLE
Emergency mashup

### PR DESCRIPTION
**Pull request by brangi**
## Description

Issues:

https://github.com/department-of-veterans-affairs/va.gov-team/issues/26173
https://github.com/department-of-veterans-affairs/va.gov-team/issues/25987

## Review instance

http://5494d5486be3589426fd04ad6525180f.review.vetsgov-internal/

## Screenshots

Screen Shot 2021-06-28 at 6 18 14 PM
Screen Shot 2021-06-28 at 6 18 37 PM
Screen Shot 2021-06-28 at 6 18 52 PM

### CODE REVIEW

**Review by mpelzsherman — CHANGES_REQUESTED** [review]
I think we need to add `LocationType.EMERGENCY_CARE` to [this array](https://github.com/department-of-veterans-affairs/vets-website/blob/be3d2bbd6f76e12126681eae9d8a21470304cff4/src/applications/facility-locator/components/PaginationWrapper.jsx).

File: src/applications/facility-locator/api/MockLocatorApi.js [hunk 1]
```diff
@@ -95,6 +99,19 @@ class MockLocatorApi {
...
+              serviceType === Object.keys(emergencyCareServices)[0]
+            ) {
+              locationsData = emergencyCareData.data;
+            }
+            if (serviceType === Object.keys(emergencyCareServices)[1]) {
+              locationsData = emergencyCareData.data.slice(0, 3);
+            }
+            if (serviceType === Object.keys(emergencyCareServices)[2]) {
+              locationsData = [emergencyCareData.data[4]];
+            }
```
Comment (mpelzsherman) [inline src/applications/facility-locator/api/MockLocatorApi.js:114]: Can this block be moved into a separate function? LIke the linter warning says, this function is getting too long/complex.

File: src/applications/facility-locator/components/search-results-items/EmergencyCareResult.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+          />
+          <div className="usa-alert-body" id="emergency-care-link">
+            <a
+              href={
+                'https://www.va.gov/COMMUNITYCARE/programs/veterans/Emergency_Care.asp'
+              }
+              target={'_/blank'}
+              onClick={() => {
+                // Record event
+                recordEvent({ event: 'cta-primary-button-click' });
```
Comment (mpelzsherman) [inline src/applications/facility-locator/components/search-results-items/EmergencyCareResult.jsx:54]: Please check w/analytics about the event name here. They probably want something more specific.

**Comment by mpelzsherman** [comment]
Great job overall! The only required change is the pagination issue - the rest are just suggestions.

**Comment by gescobardev** [comment]
Great work @brangi !

**Comment by brangi** [comment]
Clicks on the link to the emergency care benefit page are tracked (event: cta-emergency-benefit-button-click)
 Environment check is in place to hold changes in staging until validated by team and accessibility specialists

**Review by mpelzsherman — APPROVED** [review]

**Review by gescobardev — APPROVED** [review]

**Pull request merged**